### ISSUE TITLE
[React][Vite] ReferenceError: React is not defined. Disclosure component

### ISSUE THREAD

**Issue by valkedev** [tailwindlabs/headlessui#1754]
@headlessui/react v1.6.6 Chrome

There is a problem with Discloture component.

I use Vite with React and experience this kind of error:
```
index.25fddd55.js:46 ReferenceError: React is not defined
    at index.25fddd55.js:67:21695
    at Ch (index.25fddd55.js:44:57576)
    at gi (index.25fddd55.js:44:66437)
    at ck (index.25fddd55.js:48:11909)
    at bk (index.25fddd55.js:48:1039)
    at ak (index.25fddd55.js:48:962)
    at Tj (index.25fddd55.js:48:803)
    at Lj (index.25fddd55.js:46:11940)
    at index.25fddd55.js:44:41737
    at rt.unstable_runWithPriority (index.25fddd55.js:33:4130)
```

Found solution in `node_modules/@headlessui/react/dist/components/disclosure/disclosure.js` on 1:2208:
instead of `t.as===y.Fragment` i used `t.as===j` because you're already import Fragment as j in this file and problem solved for me.

Maybe i did something wrong but i really want a quick solution for that.

Thant you.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This is a vite problem, and not a Headless UI problem. However, we have an improvement that seems to solve this as well implemented here: #1693 

You should be able to use the `insiders` build to "fix" this problem.

You can already try it using `npm install @headlessui/vue@insiders`.

**Comment by valkedev** [comment]
@RobinMalfait Heya!
It helped! Thank you!
